app/src/org/commcare/connect/MessageManager.java

```diff
@@ -215,10 +215,13 @@ public static void getChannelEncryptionKey(Context context, ConnectMessagingChan
                 new IApiCallback() {
                     @Override
                     public void processSuccess(int responseCode, InputStream responseData) {
-                        ApiConnectId.handleReceivedEncryptionKey(context, responseData, channel);
-
-                        if (listener != null) {
-                            listener.connectActivityComplete(true);
+                        try (InputStream in = responseData){
+                            ApiConnectId.handleReceivedEncryptionKey(context, in, channel);
+                            if (listener != null) {
+                                listener.connectActivityComplete(true);
+                            }
+                        } catch (IOException e) {
+                            Logger.log(LogTypes.TYPE_CONNECT_MESSAGE_KEY,"Exception occurred while handling received encryption key");
```

Comment:
Our log types are pretty generic and generally are not that useful at the moment for filtering. we can just have it as `LogTypes.TYPE_EXCEPTION`